File: backend/src/main/java/com/gymmanagement/backend/dto/post/UserPostDTO.java
package com.gymmanagement.backend.dto.post;

import com.gymmanagement.backend.model.User.UserType;
import jakarta.validation.constraints.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserPostDTO {
    @NotBlank
    @Size(max = 100)
    private String name;

    @NotBlank
    private String password;

    @NotBlank
    @Email
    private String email;

    @NotBlank
    @Pattern(regexp = "^[0-9]{9}$")
    private String phone;

    @NotNull
    @Past
    private LocalDate birthDate;

    @NotNull
    private UserType userType;

    private boolean active = true;
}

File: frontend/lib/models/post/UserPostDTO.dart

import '../enums.dart';

class UserPostDTO {
  final String name;
  final String password;
  final String email;
  final String phone;
  final DateTime birthDate;
  final UserType userType;
  final bool active;

  UserPostDTO({
    required this.name,
    required this.password,
    required this.email,
    required this.phone,
    required this.birthDate,
    required this.userType,
    this.active = true,
  });

  factory UserPostDTO.fromJson(Map<String, dynamic> json) {
    return UserPostDTO(
      name: json['name'],
      password: json['password'],
      email: json['email'],
      phone: json['phone'],
      birthDate: DateTime.parse(json['birthDate']),
      userType: UserType.values.firstWhere((e) => e.name == json['userType']),
      active: json['active'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'password': password,
      'email': email,
      'phone': phone,
      'birthDate': birthDate.toIso8601String(),
      'userType': userType.name,
      'active': active,
    };
  }
}